lib/event.rb

```diff
@@ -0,0 +1,29 @@
+class Event
+  attr_reader :name, :crafts, :attendees
+  def initialize(name, crafts, attendees)
+    @name = name
+    @crafts = crafts
+    @attendees = attendees
+  end
+
+  def attendee_names
+    attendees.map(&:name)
+  end
+
+  def craft_with_most_supplies
+    most_crafts = crafts.group_by(&:name).max_by do  |_k, v|
+       v[0].supplies_required.length
+    end
+    most_crafts[0]
+  end
```

Comment:
I think this is a bit overcomplicated. I might grab the name on the craft after we have found the craft with the most supplied to eliminate the `group_by` clause. Then we are only working with an array of crafts:

```suggestion
  def craft_with_most_supplies
    most_crafts = crafts.max_by do  |craft|
       craft.supplies_required.length
    end
    most_crafts.name
  end
```